### PR TITLE
[10.0][HOOK] stock: Add picking_assign domain hook

### PR DESCRIPTION
**Pull request by rousseldenis**
Description of the issue/feature this PR addresses:

Allows to override picking search domain in stock_move.picking_assign method.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by rousseldenis** [comment]
This can be a fast merge

**Comment by rousseldenis** [comment]
@jco-odoo

**Comment by mreficent** [comment]
@sle-odoo

File: addons/stock/models/stock_move.py [hunk 1]
```diff
@@ -411,6 +411,20 @@ def onchange_date(self):
...
+        return [
+            ('group_id', '=', move.group_id.id),
+            ('location_id', '=', move.location_id.id),
+            ('location_dest_id', '=', move.location_dest_id.id),
+            ('picking_type_id', '=', move.picking_type_id.id),
+            ('printed', '=', False),
+            ('state',
+             'in',
+             ['draft', 'confirmed', 'waiting', 'partially_available',
+              'assigned'])
```
Comment (sle-odoo) [inline addons/stock/models/stock_move.py]: the indentation looks weird

File: addons/stock/models/stock_move.py [hunk 2]
```diff
@@ -421,13 +435,10 @@ def assign_picking(self):
...
-                ('group_id', '=', move.group_id.id),
-                ('location_id', '=', move.location_id.id),
-                ('location_dest_id', '=', move.location_dest_id.id),
-                ('picking_type_id', '=', move.picking_type_id.id),
-                ('printed', '=', False),
-                ('state', 'in', ['draft', 'confirmed', 'waiting', 'partially_available', 'assigned'])], limit=1)
+            picking = Picking.search(
+                self._get_picking_assign_domain(move),
+                limit=1
+            )
```
Comment (sle-odoo) [inline addons/stock/models/stock_move.py]: same here

**Comment by sle-odoo** [comment]
not against the idea, please fix the indentation

**Comment by mart-e** [comment]
Actually, it was already done (and merged) in master at #23300

**Comment by rousseldenis** [comment]
@mart-e Will it be backported in 10.0 ?

**Comment by mart-e** [comment]
@rousseldenis no, sorry but hook methods are not bug fixes and land in master, not in stable versions

**Comment by rousseldenis** [comment]
@mart-e Ok, good to know as it was not the case before as some PR's like this have been merged.

**Pull request closed without merging**